docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/BuildingsProcessor.php

```diff
@@ -65,6 +102,78 @@ protected function processRecord(&$record) {
     }
   }
 
+  /**
+   * Initialize relevant services.
+   */
+  public function init() {
+    $this->client = \Drupal::service('http_client');
+    $this->fs = \Drupal::service('file_system');
+    $this->configFactory = \Drupal::service('config.factory');
+    $this->imageFieldConfig = FieldConfig::loadByName('node', 'building', 'field_building_image');
+  }
+
+  /**
+   * Cache local image from API.
+   *
+   * Save a local version of an image gotten from the facilities API
+   * and assign that as the building image. Additionally, add title
+   * and alt text to the image field based on API data.
+   *
+   * @param array $result
+   *   The result array reference that contains the image URL.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *    Throws an exception if it can't get the image for the building.
+   */
+  protected function processResultImage(&$result) {
+    if (!empty($result->imageUrl)) {
+      try {
+        $building_image_url = $result->imageUrl;
+        $this->client->request('GET', $building_image_url);
+
+        $scheme = $this->configFactory->get('system.file')->get('default_scheme');
+        $destination = $scheme . '://' . $this->imageFieldConfig->getSetting('file_directory') . '/';
+        $building_number = $result->buildingNumber;
+        $realpath = $this->fs->realpath($destination);
+
+        if ($this->fs->prepareDirectory($realpath, FileSystemInterface::CREATE_DIRECTORY)) {
+          $data = file_get_contents($building_image_url);
+          $file = \Drupal::service('file.repository')->writeData($data, "{$destination}{$building_number}.jpg", FileSystemInterface::EXISTS_REPLACE);
+
+          // In trying to be thoughtful about the alt text,
+          // this next block intends to only construct readable, usable text
+          // while discarding unnecessary concatenators or null values.
+          $buildingName = $result?->buildingFormalName ?: '';
+          $address = $result?->address ? 'located at ' . $result?->address : NULL;
+          $address2 = $result?->address2 ? ', ' . $result?->address2 : NULL;
+          $addressFull = $address ? $address . $address2 : NULL;
+
+          $altText = $buildingName;
+          if ($altText && $addressFull) {
+            $altText .= (', ' . $addressFull);
+          }
+          else {
+            $altText = $addressFull;
+          }
+
+          $result->imageUrl = [
+            'target_id' => $file->id(),
+            'title' => $buildingName,
+            'alt' => $altText,
+          ];
```

Comment:
Current Drupal coding style guide has variables using snake-case (`$building_name`) instead of camel case (`$buildingName`). This is confusing because class properties are supposed to use camel case. So `$result->imageUrl` is fine.